proposals/NNNN-async-main-semantics.md

```diff
@@ -0,0 +1,268 @@
+# Asynchronous Main Semantics
+
+* Proposal: [SE-NNNN](NNNN-async-main-semantics.md)
+* Authors: [Evan Wilde](https://github.com/etcwilde)
+* Review Manager: TBD
+* Status: **Awaiting Implementation**
+
```

Comment:
Please add links to the pitches and implementation

Reply:
Done